R/zchunk_L152.MACC.R

```diff
@@ -26,50 +27,88 @@ module_emissions_L152.MACC_DISABLED <- function(command, ...) {
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code converts gdp using a conv_xxxx_xxxx_USD constant
-    # Use the `gdp_deflator(year, base_year)` function instead
-    # ===================================================
+    # Assign MACC data based on MACC curve year assumption (emissions.EPA_MACC_YEAR)
+    if( emissions.EPA_MACC_YEAR == 2020 ) EPA_MACC_MtCO2e <- EPA_MACC_2020_MtCO2e
+    if( emissions.EPA_MACC_YEAR == 2030 ) EPA_MACC_MtCO2e <- EPA_MACC_2030_MtCO2e
+    if( !emissions.EPA_MACC_YEAR %in% c( 2020, 2030 ) ) stop( "MAC curve year needs to be either 2020 or 2030" )
+
+    # Make processes and region names consistent
+    EPA_MACC_baselines_MtCO2e <- EPA_MACC_baselines_MtCO2e_in %>%
+      mutate(Process = sub( "\\&", "and", Process )) %>%
+      mutate(EPA_region = sub( "\\&", "and", EPA_region )) %>%
+      mutate(EPA_region = sub( "World", "Global", EPA_region )) %>%
+      mutate(EPA_region = sub( "Global Total", "Global", EPA_region ))
+
+    EPA_MACC_MtCO2e <- EPA_MACC_MtCO2e %>%
+      mutate(Process = sub( "\\&", "and", Process ))
+
+    # Convert MAC cureves to long form
+    # Convert from 2010$/tCO2e to 1990$/tC
+    L152.EPA_MACC_MtCO2e <- EPA_MACC_MtCO2e %>%
+      gather(cost_2010USD_tCO2e,reduction_MtCO2e, -Sector, -Process, -EPA_region, -EPA_region_code) %>%
+      mutate(cost_2010USD_tCO2e = as.numeric(cost_2010USD_tCO2e)) %>%
+      mutate(cost_1990USD_tCe = round(cost_2010USD_tCO2e * emissions.CONV_C_CO2 * gdp_deflator(1990, base_year = 2010), 0)) %>%
+      select(-cost_2010USD_tCO2e)
 
+    # For in baseline and abatement data:
+    # Merge aluminum and magnesium processes
+    # Put in long form
+    L152.EPA_MACC_MtCO2e <- L152.EPA_MACC_MtCO2e %>%
+      ungroup() %>%
+      mutate(Process = sub( "Primary Aluminum Production", "Aluminum and Magnesium Production", Process )) %>%
+      mutate(Process = sub( "Magnesium Manufacturing", "Aluminum and Magnesium Production", Process )) %>%
+      group_by(Sector, Process, EPA_region, EPA_region_code, cost_1990USD_tCe) %>%
+      summarize_at(vars(reduction_MtCO2e), sum)
+
+    # Repeat same operation on baseline curve
```

Comment:
group_by and summarize are both called twice, but in each instance I'm grouping by and summarizing different columns.... 
If anything the lines above it would benefit from being called inside a function. 
`      mutate(Process = sub("Primary Aluminum Production", "Aluminum and Magnesium Production", Process)) %>%
      mutate(Process = sub("Magnesium Manufacturing", "Aluminum and Magnesium Production", Process)) %>%`
Can you clarify?